website/static/js/contribTreebeard.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var ko = require('knockout');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
+
+function expandOnLoad() {
+    var tb = this;  // jshint ignore: line
+    for (var i = 0; i < tb.treeData.children.length; i++) {
+        var parent = tb.treeData.children[i];
+        tb.updateFolder(null, parent);
+        expandChildren(tb, parent.children);
+    }
+}
+
+function expandChildren(tb, children) {
+    var openParent = false;
+    for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        var parent = children[i].parent();
+        if (child.children.length > 0) {
+            expandChildren(tb, child.children);
+        }
+    }
+    if (openParent) {
+        openAncestors(tb, children[0]);
+    }
+}
+
+function openAncestors (tb, item) {
+    var parent = item.parent();
+    if(parent && parent.id > 0) {
+        tb.updateFolder(null, parent);
+        openAncestors(tb, parent);
+    }
+}
+
+function NodesPrivacyTreebeard(data, nodesState, nodesOriginal) {
```

Comment:
Looks like this file is a nearly exact copy/paste from [`nodesPrivacySettingsTreebeard.js`](https://github.com/CenterForOpenScience/osf.io/blob/ec285291ea829cf13ff47a8b3631474d35787baa/website/static/js/nodesPrivacySettingsTreebeard.js). Here's the diff:

```diff
--- nodesPrivacySettingsTreebeard.js	2016-01-05 09:57:52.000000000 -0500
+++ contribTreebeard.js	2016-01-08 12:32:46.000000000 -0500
@@ -7,7 +7,6 @@
 var $osf = require('js/osfHelpers');
 var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
 
-
 function expandOnLoad() {
     var tb = this;  // jshint ignore: line
     for (var i = 0; i < tb.treeData.children.length; i++) {
@@ -39,13 +38,13 @@
     }
 }
 
-function NodesPrivacyTreebeard(divID, data, nodesState, nodesOriginal) {
+function NodesPrivacyTreebeard(data, nodesState, nodesOriginal) {
     /**
      * nodesChanged and nodesState are knockout variables.  nodesChanged will keep track of the nodes that have
      *  changed state.  nodeState is all the nodes in their current state.
      * */
     var tbOptions = $.extend({}, projectSettingsTreebeardBase.defaults, {
-        divID: divID,
+        divID: 'grid',
         filesData: data,
         naturalScrollLimit : 0,
         rowHeight : 35,
@@ -78,7 +77,6 @@
                     filter : false,
                     custom : function () {
                         return m('input[type=checkbox]', {
-                            disabled : !item.data.node.can_write,
                             onclick : function() {
                                 item.data.node.is_public = !item.data.node.is_public;
                                 item.open = true;
@@ -114,4 +112,3 @@
     expandOnLoad.call(grid.tbController);
 }
 module.exports = NodesPrivacyTreebeard;
-
```